pkg/logql/ast.go

```diff
@@ -964,6 +962,18 @@ func (e *VectorAggregationExpr) String() string {
 
 // impl SampleExpr
 func (e *VectorAggregationExpr) Shardable() bool {
+	if e.Operation == OpTypeCount || e.Operation == OpTypeAvg {
```

Comment:
We really need a way to express operations that aren't `always` or `never` shardable, but shardable depending on how they're combined with other operations. This is a great start 👍.